server/game_components/game_objects.py

```diff
@@ -240,87 +240,68 @@ def action(self, _caster: Entity, _target: Entity | None) -> list[ActionDict]:
         :param _target: the target(s)
         :return: its a list of what happened. Single target attacks still return a list of dicts
         """
-        if _target is None:
-            assert self.requires_target is False
```

Comment:
you could shorten this to `assert not self.requires_target`. Also, consider not using `assert` here, and raise an exception instead (it can be a custom one).
```py
if self.requres_target:
    raise SomeError("blah blah")
``` 
`assert` is primarily a tool meant for use in tests, and in cases where some condition should never happen. Python will ignore `assert`s and won't compile them when you run the program with optimizations on and/or when `__debug__` is set to `False`.